lib/charms/opensearch/v0/opensearch_base_charm.py

```diff
@@ -604,31 +613,29 @@ def _is_tls_fully_configured(self) -> bool:
 
         return self._are_all_tls_resources_stored()
 
-    def _start_opensearch(self, event: EventBase) -> None:  # noqa: C901
+    def _start_opensearch(self, event: _StartOpenSearch) -> None:  # noqa: C901
         """Start OpenSearch, with a generated or passed conf, if all resources configured."""
+        if not self.node_lock.acquired:
+            logger.debug("Lock to start opensearch not acquired. Will retry next event")
+            event.defer()
+            return
         if self.opensearch.is_started():
             try:
                 self._post_start_init()
             except (OpenSearchHttpError, OpenSearchNotFullyReadyError):
                 event.defer()
             return
         if not self._can_service_start():
-            self.peers_data.delete(Scope.UNIT, "starting")
+            self.node_lock.release()
             event.defer()
             return
 
-        if self.peers_data.get(Scope.UNIT, "starting", False) and self.opensearch.is_failed():
-            self.peers_data.delete(Scope.UNIT, "starting")
+        # TODO: need to check if not first execution after acquiring lock?
+        if self.opensearch.is_failed():
```

Comment:
The `starting` flag is useful when the start process took so long that it triggered a `OpenSearchStartTimeoutError` or when the node started but is not ready yet and raised a 503 error `OpenSearchNotFullyReadyError` thus event deferred - but no other unit should start because this unit hasn't completed `starting` and a subsequent check may confirm it has indeed fully started after a few moments.